include/boost/geometry/algorithms/detail/disjoint/segment_box.hpp

```diff
@@ -36,10 +38,194 @@ namespace boost { namespace geometry
+                            CT& lon2,
+                            CT& lat2)
+    {
+        std::swap(lon1, lon2);
+        std::swap(lat1, lat2);
+    }
+
+
+    template <typename CT, typename Strategy>
+    static inline CT compute_vertex_lon(CT lon1,
+                                        CT lat1,
+                                        CT lon2,
+                                        CT lat2,
+                                        CT vertex_lat,
+                                        CT alp1,
+                                        Strategy azimuth_strategy)
+    {
+        if (vertex_lat == lat1)
+        {
+            return lon1;
+        }
+        if (vertex_lat == lat2)
+        {
+            return lon2;
+        }
+
+        return disjoint_segment_box_call_vertex_longitude<CT, CS_Tag>
+                ::apply(lat1,
+                        lat2,
+                        vertex_lat,
+                        lon2 - lon1,
+                        alp1,
+                        azimuth_strategy)
+                + lon2;
+
+    }
+
+public:
+
+    template <typename Segment, typename Box, typename Strategy>
+    static inline bool apply(Segment const& segment,
+                             Box const& box,
+                             Strategy const& azimuth_strategy)
+    {
+        assert_dimension_equal<Segment, Box>();
+
+        typedef typename point_type<Segment>::type segment_point_type;
+        typedef typename cs_tag<Segment>::type segment_cs_type;
+
+        segment_point_type p0, p1;
+        geometry::detail::assign_point_from_index<0>(segment, p0);
+        geometry::detail::assign_point_from_index<1>(segment, p1);
+
+        // Test simple case of intersection first
+
+        if (geometry::covered_by(p0, box) != geometry::covered_by(p1, box))
+        {
+            return false;
+        }
+
+        // Test intersection by comparing angles
+
+        typedef typename coordinate_type<segment_point_type>::type CT;
+
+        segment_point_type p0_normalized =
+                geometry::detail::return_normalized<segment_point_type>(p0);
+        segment_point_type p1_normalized =
+                geometry::detail::return_normalized<segment_point_type>(p1);
+
+        CT lon1 = geometry::get_as_radian<0>(p0_normalized);
+        CT lat1 = geometry::get_as_radian<1>(p0_normalized);
+        CT lon2 = geometry::get_as_radian<0>(p1_normalized);
+        CT lat2 = geometry::get_as_radian<1>(p1_normalized);
+
+        if (lon1 > lon2)
+        {
+            swap(lon1, lat1, lon2, lat2);
+        }
+
+        CT alp1, a_b0, a_b1, a_b2, a_b3;
+
+        CT b_lon_min = geometry::get<geometry::min_corner, 0>(box) * math::d2r<CT>();
+        CT b_lat_min = geometry::get<geometry::min_corner, 1>(box) * math::d2r<CT>();
+        CT b_lon_max = geometry::get<geometry::max_corner, 0>(box) * math::d2r<CT>();
+        CT b_lat_max = geometry::get<geometry::max_corner, 1>(box) * math::d2r<CT>();
```

Comment:
Ah, we've not yet a get_as_radian here... This is actually not completely right (I think) because you assume coordinates are in degrees, which is not required for this algorithm.